### PR TITLE
Return user list for specific admin with pagination

### PR DESCRIPTION
**Pull request by MishaHnatyshyn**
**Description:**  return user list for specific admin with pagination

**Link to Trello ticket:** https://trello.com/c/PqvSTDz3/37-be-implement-api-for-returning-user-list-for-specific-admin-with-pagination

### CODE REVIEW

File: src/modules/users/dto/userList.dto.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { IsNumber } from 'class-validator';
+import { Type } from "class-transformer";
+
+export class UserListDto {
+  @Type(() => Number)
```
Comment (MishaAfanasiuk) [inline src/modules/users/dto/userList.dto.ts:5]: Why do you need '@Type'

File: src/modules/users/users.controller.ts [hunk 2]
```diff
@@ -17,8 +18,8 @@ export class UsersController {
 
   @UseGuards(AuthGuard('admin'))
   @Get()
-  async findAll() {
-    return await this.usersService.findAll();
+  async findAll(@Query() userListDto: UserListDto, @Request() req) {
```
Comment (MishaAfanasiuk) [inline src/modules/users/users.controller.ts]: Why not use object destruction here or in-service method?

Comment (MishaAfanasiuk) [reply src/modules/users/users.controller.ts]: Why not remove async await?)

File: src/modules/users/dto/userList.dto.ts (hunk 1 above)
Comment (MishaHnatyshyn) [reply src/modules/users/dto/userList.dto.ts:5]: In order to convert number from string format to number format

**Review by MishaAfanasiuk — APPROVED** [review]

**Pull request merged**